src/Selenium/SeleniumServerAdapter.php

```diff
@@ -63,6 +112,84 @@ public function isSeleniumServer($seleniumServerUrl)
             return false;
         }
 
+        $this->cloudService = $this->detectCloudServiceByStatus(json_decode($responseData));
+
         return true;
     }
+
+    /**
+     * Get name of the cloud service we are connected to.
+     *
+     * @return string Cloud service identifier; empty string if no cloud service detected
+     */
+    public function getCloudService()
+    {
+        // If cloud service value is not yet initialized, attempt to connect to the server first
+        if (is_null($this->cloudService)) {
+            if (!$this->isSeleniumServer()) {
+                throw new \RuntimeException(sprintf('Unable to connect to remote server: %s', $this->getLastError()));
+            }
+        }
+
+        return $this->cloudService;
+    }
+
+    /**
+     * @param $seleniumServerUrl
+     * @return array
+     */
+    protected function parseServerUrl($seleniumServerUrl)
+    {
+        $urlParts = parse_url($seleniumServerUrl);
+
+        if (empty($urlParts['scheme']) || empty($urlParts['host'])) {
+            throw new \RuntimeException(sprintf('Provided Selenium server URL "%s" is invalid', $seleniumServerUrl));
+        }
+
+        if (empty($urlParts['port'])) {
+            if ($this->detectCloudServiceByHost($urlParts['host'])) {
+                $urlParts['port'] = self::DEFAULT_PORT_CLOUD_SERVICE;
+            } else {
+                $urlParts['port'] = self::DEFAULT_PORT;
+            }
+        }
+
+        return $urlParts;
+    }
+
+    /**
+     * Attempt to detect if given host leads to some known cloud service
+     *
+     * @param string $host
+     * @return bool
+     */
+    protected function detectCloudServiceByHost($host)
```

Comment:
I see, had similar thoughts when naming the method and even originally tried name like you propose, however, it was really long then, so I put it only in the description :)